src/common/dom/apply_themes_on_element.ts

```diff
@@ -21,26 +22,41 @@ const hexToRgb = (hex: string): string | null => {
  * element: Element to apply theme on.
  * themes: HASS Theme information
  * localTheme: selected theme.
- * updateMeta: boolean if we should update the theme-color meta element.
+ * mainElement: boolean if it is the top element of the page.
  */
 export const applyThemesOnElement = (
   element,
-  themes,
-  localTheme,
-  updateMeta = false
+  themes: HomeAssistant["themes"],
+  localTheme?: string | null,
+  mainElement = false
 ) => {
+  // We only set styles if the element has an theme itself or on the main element, otherwise it will inherit the styles from it's parent
+  let newTheme: Theme | undefined;
+  if (localTheme && themes.themes[localTheme]) {
+    newTheme = themes.themes[localTheme];
+  } else if (mainElement && localTheme !== "default") {
```

Comment:
It still feels a little weird that we have this specific mainElement logic. Instead passing if mainElement is being used, can't we just pass the default theme as the `localTheme` variable when we need to put it on main element ?